kolibri/plugins/learn/assets/src/state/actions.js

```diff
@@ -682,8 +683,14 @@ function setAndSaveCurrentExamAttemptLog(store, contentId, itemId, currentAttemp
       [itemId]: currentAttemptLog,
     }),
   });
-  const examAttemptLogModel = ExamAttemptLogResource.getModel(
-    currentAttemptLog.id);
+  let examAttemptLogModel = ExamAttemptLogResource.findModel({
+    content_id: contentId,
+    item: itemId,
+  });
+  if (!examAttemptLogModel) {
+    examAttemptLogModel = ExamAttemptLogResource.getModel(
+      currentAttemptLog.id);
+  }
```

Comment:
In short, it's because if a save happened while the first save hadn't completed, a new model would be created, so we'd end up with two attempt logs. Will add comments.